src/clr-angular/layout/tabs/tabs.ts

```diff
@@ -51,13 +54,28 @@ import { ClrCommonStrings } from '../../utils/i18n/common-strings.interface';
         </ng-container>
     `,
   providers: [IfActiveService, IfOpenService, TabsService, TABS_ID_PROVIDER],
+  host: {
+    '[class.tabs-vertical]': 'tabsService.orientation === "vertical"',
+  },
 })
 export class ClrTabs implements AfterContentInit {
-  @ContentChildren(ClrTabLink, { descendants: true })
-  tabLinkDirectives: QueryList<ClrTabLink>;
+  @Input('orientation')
+  set orientation(orientation: 'horizontal' | 'vertical') {
+    this.tabsService.orientation = orientation;
+  }
+  get orientation(): 'horizontal' | 'vertical' {
+    return this.tabsService.orientation;
+  }
 
-  @ContentChildren(ClrTabContent, { descendants: true })
-  tabContents: QueryList<ClrTabContent>;
+  @ContentChildren(ClrTab) private tabs: QueryList<ClrTab>;
+
+  get tabLinkDirectives(): ClrTabLink[] {
+    return this.tabs.map(tab => tab.tabLink);
+  }
+
+  get tabContents(): ClrTabContent[] {
+    return this.tabs.filter(tab => !!tab.tabContent).map(tab => tab.tabContent);
```

Comment:
TBD

Reply:
Actually, this array is pretty fine as it is. Due to the *clrIfActive it is far more dynamic than the clrTabLinks case. Now this call is carried only on active tab change, which is the desired behavior. It is not feasible to try to cover the case without *clrIfActive, the mixed case, etc.

Reply:
I would need to inspect this again more closely, but can't we just get this anytime the `tabs.changes` subscription fires like the link directives? It might be firing now only when tabs change, (for now) but this is still a public property that could be optimized if I'm reading it right. I think we'd rather put it in the right place and ensure it fires on tab changes, unless that doesn't function.

Reply:
This is my last question, just wanting to bring it back to attention.

Reply:
To cover the `*clrIfActive` case we have to subscribe not only to `tabs.changes`, but to `IfActiveService#currentChange`. It might be possible to make some optimization for the `no-clrIfActive` case, but the tabs component will need to know if `*clrIfActive` has been applied to it. I'm still not sure that the added complexity and risk for breaking corner cases is worth the optimization of a possible, though improbable, future case.
Also the way that the tabs component is used does not imply neither huge amounts of tabs, nor frequent repetitive calls of this method.
These are my reasons to prefer a mildly suboptimal approach, which is safer and easier to maintain.